In [3]:
import pandas as pd
import pickle

#### all_note_data에서 중복값 정리

In [7]:
all_notes = pd.read_csv('../data/all_note_data.csv')
notes_info = all_notes.iloc[:,1:]  # rows : 973
notes_info[notes_info['note'] == 'chinotto']

note  category_idx  category_name        season
211  chinotto            11      beverages  [0, 1, 2, 3]
212  chinotto             1  citrus_smells        [0, 1]

In [27]:
all_notes.head()

Unnamed: 0             note  category_idx               category_name  \
0           0         absinthe            11                   beverages   
1           1           acacia             3                     flowers   
2           2       acai berry             2  fruits_vegetables_and_nuts   
3           3          acerola             2  fruits_vegetables_and_nuts   
4           4  acerola blossom             3                     flowers   

         season  
0  [0, 1, 2, 3]  
1  [0, 1, 2, 3]  
2        [0, 2]  
3        [0, 2]  
4  [0, 1, 2, 3]

In [8]:
duplicate = notes_info[notes_info.duplicated(subset=['note'])]

In [35]:
duplicate.index

Int64Index([212, 250, 260, 612], dtype='int64')

In [16]:
notes_info[notes_info['note'] == 'chinotto']

note  category_idx  category_name        season
211  chinotto            11      beverages  [0, 1, 2, 3]
212  chinotto             1  citrus_smells        [0, 1]

In [19]:
notes_info = notes_info.drop(211)
notes_info[notes_info['note'] == 'chinotto']

note  category_idx  category_name  season
212  chinotto             1  citrus_smells  [0, 1]

In [10]:
notes_info[notes_info['note'] == 'cognac']

note  category_idx category_name        season
249  cognac            11     beverages  [0, 1, 2, 3]
250  cognac            11     beverages  [0, 1, 2, 3]

In [15]:
notes_info = notes_info.drop(249)
notes_info[notes_info['note'] == 'cognac']

note  category_idx category_name        season
250  cognac            11     beverages  [0, 1, 2, 3]

In [20]:
notes_info[notes_info['note'] == 'cork']

note  category_idx                            category_name        season
259  cork             8                         woods_and_mosses        [2, 3]
260  cork            12  netural_and_synthetic_popular_and_weird  [0, 1, 2, 3]

In [23]:
#notes_info = notes_info.drop(260)
notes_info[notes_info['note'] == 'cork']

note  category_idx     category_name  season
259  cork             8  woods_and_mosses  [2, 3]

In [24]:
notes_info[notes_info['note'] == 'molasses']

note  category_idx                            category_name  \
611  molasses            12  netural_and_synthetic_popular_and_weird   
612  molasses             7               sweets_and_gourmand_smells   

           season  
611  [0, 1, 2, 3]  
612           [3]

In [26]:
#notes_info = notes_info.drop(611)
notes_info[notes_info['note'] == 'molasses']

note  category_idx               category_name season
612  molasses             7  sweets_and_gourmand_smells    [3]

In [28]:
notes_info.to_csv('../data/all_note_data.csv', index=False)

#### 그에 따른 notes의 pk값 수정
* chinotto : [211, 212] --> 212
* cognac : [249, 250] --> 250
* cork : [259, 260] --> 259
* molasses : [611, 612] --> 612

In [30]:
with open('../data/notes.pkl', 'rb') as f:
    notes_dicts = pickle.load(f)

In [32]:
notes_dicts[212]

{'pk': 213,
 'model': 'perfumes.note',
 'fields': {'name': 'chinotto', 'kor_name': '치노토'}}

In [34]:
notes_dicts[211]

{'pk': 212,
 'model': 'perfumes.note',
 'fields': {'name': 'chinotto', 'kor_name': '치노토'}}